modules/dependencies.nix

```diff
@@ -0,0 +1,30 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.dependencies;
+
+  packages = {
+    inherit (pkgs)
+      curl
+      git
+      ;
+    ueberzug = pkgs.ueberzugpp;
+  };
+
+  mkDependencyOption = name: default: {
+    enable = lib.mkEnableOption "Add ${name} to dependencies.";
+
+    package = lib.mkPackageOption pkgs name { };
+  };
+in
+{
+  options.dependencies = lib.mapAttrs mkDependencyOption packages;
+
+  config = {
+    extraPackages = lib.mapAttrsToList (_: p: if p.enable then p.package else null) cfg;
```

Comment:
Feels simpler to filter on `enable`? Rather than adding a bunch of null definitions?

```nix
extraPackages = lib.pipe cfg [
  (lib.filterAttrs (_: dep: dep.enable))
  (builtins.catAttrs "package")
  # or (builtins.map (dep: dep.package))
]
```